src/graphnet/data/curated_datamodule.py

```diff
@@ -0,0 +1,282 @@
+"""Contains a Generic class for curated DataModules/Datasets.
+
+Inheriting subclasses are data-specific implementations that allow the user to
+import and download pre-converteddatasets for training of deep learning based
+methods in GraphNeT.
+"""
+
+from typing import Dict, Any, Optional, List, Tuple, Union
+from abc import abstractmethod
+import os
+
+from .datamodule import GraphNeTDataModule
+from graphnet.models.graphs import GraphDefinition
+from graphnet.data.dataset import ParquetDataset, SQLiteDataset
+
+
+class CuratedDataset(GraphNeTDataModule):
+    """Generic base class for curated datasets.
+
+    Curated Datasets in GraphNeT are pre-converted datasets that have been
+    prepared for training and evaluation of deep learning models. On these
+    Datasets, graphnet users can train and benchmark their models against SOTA
+    methods.
+    """
+
+    def __init__(
+        self,
+        graph_definition: GraphDefinition,
+        download_dir: str,
+        truth: Optional[List[str]] = None,
+        features: Optional[List[str]] = None,
+        backend: str = "parquet",
```

Comment:
Are we slowly changing over to the default backend being parquet? Or is it just that the example dataset here is parquet? I was under the impression that sqlite was currently the "default" and what most people used?

Reply:
There is not really much thought behind this default value. It's not intended to be read as a suggestion on what we recommend people to use. The code becomes a bit simpler if we allow for this argument to default to one of the backends, and by chance I chose `parquet`.

The parquet backend takes up ~8x less space and has a constant query speed, making it the "recommended" backend for very large datasets, or for people with tight space quotas. Because it does not offer fast random access to the entire dataset, it's somewhat difficult to work with, and is thus a bit more advanced.

sqlite has the fastest query speeds for small-medium sized datasets, and has full random access, but takes up quite a bit of space, and is therefore not scalable into very large datasets. It is quite easy to work with.